tests/schema/test_union.py

```diff
@@ -662,3 +662,29 @@ class Query:
         union: Union[Something, AnnotatedInt]
 
     strawberry.Schema(query=Query)
+
+
+@pytest.mark.parametrize(
+    "annotation", ("int", "str", "float", "list[str]", "List[str]")
+)
+def test_raises_on_union_of_scalars(annotation: str, monkeypatch):
+    # using forward refs because they deduced later
+    # catching all edge cases.
+    if annotation == "list[str]" and sys.version_info < (3, 9, 0):
+        pytest.skip("list has not __class_getitem__ prior to py39")
```

Comment:
This can be placed in the parametrize on line 668 using `pytest.param("list[str", marks=pytest.mark.skipif(...))` 😄